# Profitable App Profiles for the App Store and Google Play Markets



.سودآور باشد Google Play و App Store  هدف ما در این پروژه یافتن ایده ای برای ساخت یک برنامه تلفن همراه است که در فروشگاه های  

.را می سازد iOS  و Android  به عنوان یک تحلیلگر داده برای شرکتی کار می کنیم که برنامه های تلفن همراه
<br>
.و وظیفه ما این است که تیم توسعه دهندگان خود را قادر سازیم تا با استدلال مبتنی بر تحلیل داده ، برنامه ای سودمند بسازند
<br><br>
در شرکت ما، تنها برنامه هایی ساخته می شود که به صورت رایگان دانلود و نصب می شوند و منبع اصلی درآمد ما، تبلیغات درون برنامه ای است؛ بدین معنا که درآمد ما از هر برنامه بیشتر تحت تأثیر تعداد کاربرانی است که از برنامه استفاده می کنند. هدف ما از این پروژه تجزیه و تحلیل داده هاست تا به توسعه دهندگان ما کمک کند به این موضوع پی ببرند که چه نوع برنامه هایی احتمالاً باعث جذب بیشتر کاربران می شوند

## Opening and Exploring the Data


موجود بوده است Google Play  و 2.1 میلیون برنامه آندروید در App Store در iOS  تا سپتامبر 2018 ، تقریباً 2 میلیون برنامه 
<br><br>
جمع آوری داده ی بیش از چهار میلیون برنامه نیازمند مقدار قابل توجهی زمان و هزینه است، بنابراین در عوض سعی خواهیم کرد نمونه ای از داده ها را تجزیه و تحلیل کنیم. برای جلوگیری از صرف منابع برای جمع آوری داده های جدید ، ابتدا باید ببینیم که آیا می توانیم داده های موجود مرتبط را بدون هیچ هزینه ای پیدا کنیم. خوشبختانه ، دو دیتاست وجود دارد که برای هدف ما مناسب به نظر می رسد



- [google play data set](https://pythoncoding.ir/Downloads/googleplaystore.csv) 
- [ios data set](https://pythoncoding.ir/Downloads/AppleStore.csv)


بیایید با باز کردن هر دو دیتاست شروع کرده و سپس به بررسی داده ها بپردازیم

In [1]:
from csv import reader

### The Google Play data set ###
opened_file = open('googleplaystore.csv')
read_file = reader(opened_file)
android = list(read_file)
android_header = android[0]
android = android[1:]

### The App Store data set ###
opened_file = open('AppleStore.csv')
read_file = reader(opened_file)
ios = list(read_file)
ios_header = ios[0]
ios = ios[1:]

می نویسیم تا بتوانیم به طور مکرر از آن برای بررسی سطر ها به صورت خواناتر استفاده کنیم explore_data() برای سهولت بررسی در این دو دیتاست ، ابتدا تابعی به نام 
<br>
همچنین آپشنی برای تابع خود اضافه خواهیم کرد تا تعداد سطرها و ستون ها را برای هر دیتاستی نشان دهد

In [2]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line between rows
        
    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

print(android_header)
print('\n')
explore_data(android, 0, 3, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13


می بینیم که دیتاست گوگل پلی  دارای 10841 برنامه و 13 ستون است

 با یک نگاه سریع ، می توان فهمید که ستون های زیر می توانند برای هدف تجزیه و تحلیل ما مفید باشند
 
 `'App'`, `'Category'`, `'Reviews'`, `'Installs'`, `'Type'`, `'Price'`, و `'Genres'`
 
 حال بیایید نگاهی به دیتاست اپ استور بیندازیم.




In [3]:
print(ios_header)
print('\n')
explore_data(ios, 0, 3, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 7197
Number of columns: 16


داریم و ستون هایی که جالب به نظر می رسند عبارتند از iOS  در این دیتاست 7197 برنامه 

`'track_name'`, `'currency'`, `'price'`, `'rating_count_tot'`, `'rating_count_ver'`, `'prime_genre'`


نام همه ستون ها به راحتی  قابل فهم نیست، اما جزئیات مربوط به هر ستون را می توان در داکیومنتیشن دیتاست در زیر یافت

[documentation](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/home).


## Deleting Wrong Data

دارای بخش بحث و تبادل نظر است Google Play دیتاست 

[discussion section](https://www.kaggle.com/lava18/google-play-store-apps/discussion)

و می توان دید که در یکی از بحث ها

[dar in link](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015) 

کاربری خطایی را در سطر 10472 یافته است. بیایید این سطر را پرینت کنیم و آن را با هدر و یک سطر دیگر که درست است مقایسه کنیم.

In [4]:
print(android[10472])  # incorrect row
print('\n')
print(android_header)  # header
print('\n')
print(android[0])      # correct row

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


است Photo Made WI-Fi Touchscreen Photo Frame سطر 10472 متعلق به برنامه 

و می بینیم که امتیازش  19 است که کاملا غلط است؛ زیرا بیشترین امتیاز اپلیکیشن ها در گوگل پلی برابر 5 است 

است Category  این مسئله به دلیل وجود نداشتن هیچگونه مقداری در ستون 

در نتیجه ، این سطر را حذف خواهیم کرد



In [5]:
print(len(android))
del android[10472]  # don't run this more than once
print(len(android))

10841
10840


## Removing Duplicate Entries

### Part One

را به اندازه کافی بررسی کنیم ، متوجه خواهیم شد که برخی از برنامه ها بیش از یک انتری دارند Google Play  اگر دیتاست 

دارای چهار انتری است Instagram   به عنوان مثال ، برنامه


In [6]:
for app in android:
    name = app[0]
    if name == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


در کل ، 1171 مورد وجود دارد که برنامه ای  بیش از یک بار وجود دارد


In [7]:
duplicate_apps = []
unique_apps = []

for app in android:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)
    
print('Number of duplicate apps:', len(duplicate_apps))
print('\n')
print('Examples of duplicate apps:', duplicate_apps[:15])

Number of duplicate apps: 1181


Examples of duplicate apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


هنگام تجزیه و تحلیل داده ها ، نمی خواهیم برنامه ای را بیش از یک بار بشماریم ، بنابراین باید انتری های تکراری را حذف کنیم و فقط یک انتری در هر برنامه نگه داریم. کاری که می توانیم انجام دهیم ، حذف ردیف های تکراری به صورت تصادفی است، اما راه بهتری نیز وجود دارد



اگر سطر هایی را که در  دو سلول بالاتر برای برنامه اینستاگرام پرینت کردیم بررسی کنید ، تفاوت اصلی در موقعیت چهارم هر سطر اتفاق می افتد که مربوط به تعداد بازخورد هاست. اعداد مختلف نشان می دهند که داده ها در زمان های مختلفی جمع آوری شده اند. می توانیم از این ستون به عنوان معیاری جهت نگه داشتن سطر ها استفاده کنیم. سطر ها را به طور تصادفی حذف نمی کنیم ، بلکه سطر هایی را که بیشترین بازخورد را دارند حفظ خواهیم کرد زیرا هرچه تعداد بازخورد بیشتر باشد ، امتیازات قابل قبول تر است


:برای انجام این کار 

  الف) یک دیکشنری ایجاد می کنیم که در آن هر کلید دیکشنری ، نام برنامه ای یکتاست و مقدار آن، بالاترین تعداد بازخورد آن برنامه است
<br>
ب) از این دیکشنری برای ایجاد دیتاستی جدید استفاده کنید ، که فقط یک انتری برای هر برنامه دارد (و تنها برنامه های دارای بیشترین تعداد بازخورد را انتخاب می کنیم)


### Part Two

بیایید ساخت این دیکشنری را آغاز کنیم

In [8]:
reviews_max = {}

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
        
    elif name not in reviews_max:
        reviews_max[name] = n_reviews

در سلول قبل، متوجه شدیم که 1181 مورد وجود دارد که برنامه ای بیش از یک بار وجود دارد، بنابراین طول دیکشنری ما (برنامه های یکتا) باید برابر با طول دیتاست  منهای 1118 باشد.

In [9]:
print('Expected length:', len(android) - 1181)
print('Actual length:', len(reviews_max))

Expected length: 9659
Actual length: 9659


برای حذف موارد تکراری استفاده کنیم `reviews_max` حال ، بیایید از دیکشنری  

در مورد موارد تکراری ، فقط انتری های دارای بیشترین تعداد بازخورد را حفظ خواهیم کرد. در سلول زیر:

کار را شروع می کنیم `already_added`  و `android_clean`   با ساخت دو لیست خالی 

:از حلقه استفاده می کنیم  و برای هر تکرار `android` روی دیتاست 

 نام برنامه و تعداد بازخورد ها را جدا می کنیم

اضافه می کنیم و  `android_clean` را به لیست  (`app`)سطر فعلی

:اضافه می کنیم، در صورتی که(`already_added`)  را به لیست (`name`) نام برنامه 



مطابقت داشته باشد و `reviews_max`  تعداد بازخورد های برنامه فعلی با تعداد بازخورد های آن برنامه در دیکشنری  


اضافه نشده باشد  `already_added`  نام برنامه از قبل در لیست 


ما باید این شرایط تکمیلی را اضافه کنیم تا مواردی را که در آن برنامه ای دارای چند انتری است و



 
 
 بیش از 2 انتری آن دارای  بیشترین تعداد بازخورد هستند را در نظر بگیریم


دارای سه انتری است و تعداد بازخورد ها یکسان است Box   به عنوان مثال ، برنامه 

را اعمال کنیم، باز هم با انتری های تکراری برای برخی از برنامه ها مواجه خواهیم شد  `reviews_max[name] == n_reviews` اگر ما فقط شرط  
    

In [10]:
android_clean = []
already_added = []

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    
    if (reviews_max[name] == n_reviews) and (name not in already_added):
        android_clean.append(app)
        already_added.append(name) # make sure this is inside the if block

حال بیایید به سرعت دیتاست جدید را بررسی کرده و مطمین شویم که تعداد ردیف ها 9659 ردیف است

In [11]:
explore_data(android_clean, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9659
Number of columns: 13


همانطور که انتظار می رفت 9659 ردیف داریم

## Removing Non-English Apps

### Part One

اگر دیتاست را به اندازه کافی بررسی کنید، متوجه خواهید شد که نام برخی از برنامه ها به زبان انگلیسی نیست. در زیر ، چند نمونه از هر دو دیتاست را می بینیم:

In [12]:
print(ios[813][1])
print(ios[6731][1])

print(android_clean[4412][0])
print(android_clean[7940][0])

爱ه¥‡è‰؛PPS -م€ٹو¬¢ن¹گ颂2م€‹ç”µè§†ه‰§çƒ­و’­
م€گ脱ه‡؛م‚²مƒ¼مƒ م€‘絶ه¯¾مپ«وœ€ه¾Œمپ¾مپ§مƒ—مƒ¬م‚¤مپ—مپھمپ„مپ§ م€œè¬ژ解مپچ＆مƒ–مƒ­مƒƒم‚¯مƒ‘م‚؛مƒ«م€œ
ن¸­ه›½èھ‍ AQمƒھم‚¹مƒ‹مƒ³م‚°
ظ„ط¹ط¨ط© طھظ‚ط¯ط± طھط±ط¨ط­ DZ


ما علاقه ای به نگهداری این نوع برنامه ها نداریم ، بنابراین آنها را حذف خواهیم کرد

یکی از راه های انجام این کار حذف هر برنامه ای است که نام آن حاوی نمادی است که معمولاً در متن انگلیسی استفاده نمی شود

متن انگلیسی معمولاً شامل حروف الفبای انگلیسی ، اعداد بین 0 تا 9 ، علائم نگارشی (. ! ? ; و غیره) ، و سایر نمادها (+  *  / و غیره).


رمزگذاری می شوند ASCII  همه این کاراکترها که مخصوص متون انگلیسی هستند با استفاده از استاندارد 

 دارای یک عدد متناظر با آن  بین 0 تا 127 است و ما می توانیم از این مزیت برای ساخت تابعی استفاده کرده ASCII  هر کاراکتر
 
 است یا خیر ASCII  تا نام برنامه ها را بررسی و پی بریم که آیا حاوی کاراکتر های

استفاده می کنیم "ord ()" این تابع را در زیر ایجاد خواهیم کرد و برای یافتن شماره رمزگذاری مربوط به هر کاراکتر از تابع از پیش تعریف شده ی 

In [13]:
def is_english(string):
    
    for character in string:
        if ord(character) > 127:
            return False
    
    return True

print(is_english('Instagram'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))

True
False


هستند ASCII  به نظر می رسد تابع به درستی کار می کند ، اما برخی نام های برنامه های انگلیسی از اموجی یا نمادهای دیگر (™ — –  و غیره) استفاده می کنند که خارج از محدوده

به همین دلیل ، اگر از تابع فعلی استفاده کنیم ، برنامه های مفید  حذف خواهند شد


In [14]:
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

print(ord('™'))
print(ord('😜'))

False
False
8482
128540


### Part Two

داشته باشد ، برنامه را حذف خواهیم کرد ASCII  برای به حداقل رساندن تأثیر از دست رفتن داده ها ، فقط درصورتی که برنامه ای بیش از سه کاراکتر غیر

In [15]:
def is_english(string):
    non_ascii = 0
    
    for character in string:
        if ord(character) > 127:
            non_ascii += 1
    
    if non_ascii > 3:
        return False
    else:
        return True

print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

True
True


تابع هنوز بی نقص نیست و ممکن است تعداد کمی از برنامه های غیر انگلیسی از فیلتر ما عبور کنند ، اما به نظر می رسد تا به اینجا به اندازه کافی خوب عمل کرده ایم

در این مرحله نباید زمان زیادی را برای بهینه سازی صرف کنیم.

برای فیلتر کردن برنامه های غیر انگلیسی برای هر دو دیتاست استفاده می کنیم `is_english()`   در زیر ، از تابع   

In [16]:
android_english = []
ios_english = []

for app in android_clean:
    name = app[0]
    if is_english(name):
        android_english.append(app)
        
for app in ios:
    name = app[1]
    if is_english(name):
        ios_english.append(app)
        
explore_data(android_english, 0, 3, True)
print('\n')
explore_data(ios_english, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9500
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+'

باقی مانده است iOS  مشاهده می کنیم که 9614 برنامه اندروید و 6183 برنامه


## Isolating the Free Apps

همانطور که در مقدمه ذکر کردیم ، ما فقط برنامه هایی را می سازیم که دانلود و نصب رایگان دارند و منبع اصلی درآمد ما تبلیغات درون برنامه ای است

دیتاست های ما شامل برنامه های رایگان و غیر رایگان است و برای تحلیل باید فقط برنامه های رایگان را جدا کنیم. در زیر ، برنامه های رایگان را برای هر دو دیتاست جدا می کنیم


In [17]:
android_final = []
ios_final = []

for app in android_english:
    price = app[7]
    if price == '0':
        android_final.append(app)
        
for app in ios_english:
    price = app[4]
    if price == '0.0':
        ios_final.append(app)
        
print(len(android_final))
print(len(ios_final))

8760
3169


باقیمانده ایم که باید برای تجزیه و تحلیل ما کافی باشد iOS  ما با 8864 برنامه اندروید و 3222 برنامه 


## Most Common Apps by Genre

### Part One

همانطور که در مقدمه ذکر کردیم ، هدف ما تعیین ژانر برنامه هایی است که احتمالاً کاربران بیشتری را به خود جذب می کنند

 زیرا درآمد ما به شدت تحت تأثیر تعداد افرادی است که از برنامه های ما استفاده می کنند



برای به حداقل رساندن اشتباهات، استراتژی اعتبار سنجی ما برای ایده برنامه مان از سه مرحله تشکیل شده است:

اضافه کنیم Google Play  الف) حداقل نسخه آندروید برنامه را ساخته و آن را به 

ب) اگر برنامه پاسخ خوبی از سوی کاربران گرفت، آن را بیشتر توسعه دهیم.

اضافه می کنیم. App Store  برنامه را نیز می سازیم و آن را به iOS  ج) اگر برنامه بعد از شش ماه سودآور شد، نسخه 

<br>

از آنجا که هدف نهایی ما افزودن برنامه مان در اپ استور و گوگل پلی است ، باید خصوصیات برنامه ای را پیدا کنیم که در هر دو بازار موفق باشد

به عنوان مثال ، ویژگی ای که ممکن است برای هر دو بازار خوب کار کند، ممکن است یک برنامه ای باشد که از گیمیفیکیشن استفاده می کند

<br>

بیایید تجزیه و تحلیل مان را با پی بردن به متداول ترین هر بازار آغاز کنیم

ایجاد خواهیم کرد Google Play  در   دیتاست   `Category` و `Genres` و ستون های   App Store  دیتاست   `prime_genre` بدین منظور ، یک جدول فرکانسی برای ستون 



### Part Two

ما دو تابع ایجاد خواهیم کرد که می توانیم از آنها برای تجزیه و تحلیل جداول فرکانسی استفاده کنیم


الف)یک تابع برای تولید جداول فرکانسی که درصد ها را نشان می دهد

ب) تابعی دیگر که می توانیم از آن  برای نمایش درصدها به ترتیب نزولی استفاده کنیم

In [18]:
def freq_table(dataset, index):
    table = {}
    total = 0
    
    for row in dataset:
        total += 1
        value = row[index]
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
    
    table_percentages = {}
    for key in table:
        percentage = (table[key] / total) * 100
        table_percentages[key] = percentage 
    
    return table_percentages


def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
        
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

### Part Three

شروع می کنیم App Store  دیتاست `prime_genre` با بررسی جدول فرکانسی ستون 


In [19]:
display_table(ios_final, -5)

Games : 58.53581571473651
Entertainment : 7.82581255916693
Photo & Video : 5.0489113284947935
Education : 3.72357210476491
Social Networking : 3.2817923635216157
Shopping : 2.5244556642473968
Utilities : 2.398232881035027
Sports : 2.1773430104133795
Music : 2.0511202272010096
Health & Fitness : 1.9880088355948247
Productivity : 1.7040075733669928
Lifestyle : 1.5462290943515304
News : 1.3253392237298833
Travel : 1.1360050489113285
Finance : 1.1044493531082362
Weather : 0.8520037866834964
Food & Drink : 0.8204480908804039
Reference : 0.5364468286525718
Business : 0.5364468286525718
Book : 0.3786683496371095
Navigation : 0.18933417481855475
Medical : 0.18933417481855475
Catalogs : 0.12622278321236985


مشاهده می کنیم که در میان برنامه های رایگان انگلیسی ، بیش از نیمی از آنها (٪58.16) بازی هستند

برنامه های سرگرمی نزدیک به ٪8 و به دنبال آن برنامه های عکس و ویدیو هستند که نزدیک به ٪5 هستند.

تنها 3.66٪ برنامه ها برای آموزش طراحی شده اند و به دنبال آن برنامه های شبکه های اجتماعی که ٪3.29 برنامه ها را در دیتاست ما تشکیل می دهند.

<br>

تصور عمومی این است که اپ استور (حداقل بخشی که شامل برنامه های رایگان انگلیسی است) تحت سلطه برنامه هایی است 

که برای سرگرمی (بازی ، سرگرمی ، عکس و ویدئو ، شبکه های اجتماعی ، ورزش ، موسیقی و غیره) طراحی شده اند 

 در حالی که برنامه های با اهداف  کاربردی (آموزش ، خرید ، بهره وری ، سبک زندگی و غیره) نادرتر هستند
 
 با این حال ، این واقعیت که بیشترین تعداد برنامه ها، برنامه های سرگرمی هستند، به این معنی نیست که آنها بیشترین تعداد کاربر را دارند - ممکن است تقاضا همانند عرضه نباشد.
 
 <BR>

(دو ستونی که به نظر می رسد مرتبط هستند) ادامه دهیم. Google Play  از دیتاست   `Category`   و `Genres` بیایید با بررسی ستون های  

    

In [20]:
display_table(android_final, 1) # Category

FAMILY : 18.938356164383563
GAME : 9.657534246575343
TOOLS : 8.481735159817351
BUSINESS : 4.646118721461187
PRODUCTIVITY : 3.9383561643835616
LIFESTYLE : 3.9155251141552516
FINANCE : 3.721461187214612
MEDICAL : 3.550228310502283
SPORTS : 3.3333333333333335
PERSONALIZATION : 3.287671232876712
COMMUNICATION : 3.2534246575342465
HEALTH_AND_FITNESS : 3.093607305936073
PHOTOGRAPHY : 2.9794520547945202
NEWS_AND_MAGAZINES : 2.808219178082192
SOCIAL : 2.6484018264840183
TRAVEL_AND_LOCAL : 2.3401826484018264
SHOPPING : 2.2488584474885847
BOOKS_AND_REFERENCE : 2.146118721461187
DATING : 1.860730593607306
VIDEO_PLAYERS : 1.8036529680365299
MAPS_AND_NAVIGATION : 1.3812785388127853
FOOD_AND_DRINK : 1.2328767123287672
EDUCATION : 1.1757990867579908
ENTERTAINMENT : 0.9589041095890412
AUTO_AND_VEHICLES : 0.9246575342465754
LIBRARIES_AND_DEMO : 0.9018264840182649
WEATHER : 0.7876712328767124
HOUSE_AND_HOME : 0.7876712328767124
EVENTS : 0.7191780821917808
ART_AND_DESIGN : 0.6506849315068494
PARENTING : 

به طور قابل توجهی متفاوت به نظر می رسند: برنامه های زیادی برای سرگرمی طراحی نشده اند Google Play  این نتایج در 

و به نظر می رسد تعداد زیادی برنامه برای اهداف کاربردی (خانواده ، ابزار ، تجارت ، سبک زندگی ، بهره وری و غیره) طراحی شده اند

با این حال ، اگر این مورد را بیشتر بررسی کنیم ، می توانیم بفهمیم که گروه خانواده (که تقریباً ٪19 برنامه ها را تشکیل می دهد) بیشتر به معنی بازی برای بچه ها است


<BR>



[![img](https://s3.amazonaws.com/dq-content/350/py1m8_family.png)](https://play.google.com/store/apps/category/FAMILY?hl=en)


از آن خود کرده اند App Store  درصد بیشتری را  در مقایسه با  Google Play   حتی در این صورت ، به نظر می رسد برنامه های کاربردی موجود در 
    
    مشاهده می کنیم تأیید می کند `Genres`  این تصویر ، جدول فرکانسی را که برای ستون  
    
    

In [21]:
display_table(android_final, -4)

Tools : 8.470319634703197
Entertainment : 6.084474885844749
Education : 5.3881278538812785
Business : 4.646118721461187
Productivity : 3.9383561643835616
Lifestyle : 3.904109589041096
Finance : 3.721461187214612
Medical : 3.550228310502283
Sports : 3.4018264840182644
Personalization : 3.287671232876712
Communication : 3.2534246575342465
Action : 3.105022831050228
Health & Fitness : 3.093607305936073
Photography : 2.9794520547945202
News & Magazines : 2.808219178082192
Social : 2.6484018264840183
Travel & Local : 2.328767123287671
Shopping : 2.2488584474885847
Books & Reference : 2.146118721461187
Simulation : 2.054794520547945
Dating : 1.860730593607306
Arcade : 1.82648401826484
Video Players & Editors : 1.7808219178082192
Casual : 1.7351598173515983
Maps & Navigation : 1.3812785388127853
Food & Drink : 1.2328767123287672
Puzzle : 1.141552511415525
Racing : 1.004566210045662
Role Playing : 0.9474885844748858
Strategy : 0.9246575342465754
Auto & Vehicles : 0.9246575342465754
Libraries &

کاملاً واضح نیست  `Category`  و `Genres`  تفاوت ستون های 

 بسیار دانه ای تراست (دسته بندی های بیشتری دارد)  `Genres` اما نکته ای که می توانیم مشاهده کنیم این است که ستون 
 
  از این پس کار خواهیم کرد `Category` ما در حال حاضر فقط به دنبال تصویر کلی هستیم ، بنابراین فقط با ستون  


تا به اینجا، متوجه شدیم که اپ استور توسط برنامه های طراحی شده جهت سرگرمی احاطه شده است 

در حالی که گوگل پلی از منظر برنامه های کاربردی و سرگرم کننده تعادل بیشتری را نشان می دهد. اکنون می خواهیم در مورد نوع برنامه هایی که بیشترین کاربر را دارند ایده بگیریم



## Most Popular Apps by Genre on the App Store

یکی از راه های شناخت محبوب ترین ژانر(ژانر دارای بیشترین کاربر) محاسبه میانگین تعداد نصب برای هر ژانر است

پیدا کنیم ، اما برای دیتاست اپ استور این اطلاعات وجود ندارد `Installs` برای دیتاست گوگل پلی، می توانیم این اطلاعات را در ستون 

به همین دلیل، تعداد کل امتیازات داده شده توسط کاربران را به عنوان راه حل جانبی در نظر خواهیم گرفت 

مشاهده کنیم  `rating_count_tot`  که می توانیم آن را در ستون 





در زیر ، میانگین تعداد امتیازات کاربران را در هر ژانر در اپ استور محاسبه می کنیم:

In [22]:
genres_ios = freq_table(ios_final, -5)

for genre in genres_ios:
    total = 0
    len_genre = 0
    for app in ios_final:
        genre_app = app[-5]
        if genre_app == genre:            
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
    avg_n_ratings = total / len_genre
    print(genre, ':', avg_n_ratings)

Social Networking : 72916.54807692308
Photo & Video : 28441.54375
Games : 22985.211320754715
Music : 58205.03076923077
Reference : 79350.4705882353
Health & Fitness : 24037.634920634922
Weather : 54215.2962962963
Utilities : 19900.473684210527
Travel : 31358.5
Shopping : 27816.2
News : 21750.071428571428
Navigation : 86090.33333333333
Lifestyle : 16739.34693877551
Entertainment : 14364.774193548386
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 46384.916666666664
Finance : 32367.02857142857
Education : 7003.983050847458
Productivity : 21799.14814814815
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


قرار دارد  Google Maps  و Waze  به طور متوسط ، برنامه های ناوبری بیشترین تعداد بازخورد را دارند ، اما این رقم به شدت تحت تأثیر 

که در مجموع  نزدیک به نیم میلیون بازخورد را دارا هستند



In [23]:
for app in ios_final:
    if app[-5] == 'Navigation':
        print(app[1], ':', app[5]) # print name and number of ratings

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocachingآ® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


و غیره است.  Facebook ، Pinterest و Skype  همین الگو برای برنامه های شبکه های اجتماعی نیز صادق است، به طوری که تعداد میانگین به شدت تحت تأثیر چند غول فنآوری مانند 

به شدت بر روی عدد متوسط تاثیر گذاشته اند Shazam  ، Pandora  و Spotify  همین مورد در برنامه های موسیقی نیز صادق است، به طوری که چندین بازیکن بزرگ مانند 


هدف ما یافتن ژانرهای محبوب است ، اما برنامه های ناوبری ، شبکه های اجتماعی یا موسیقی ممکن است محبوب تر از آنچه که هستند به نظر برسند

به نظر می رسد میانگین تعداد  امتیازات توسط تعداد کمی از برنامه ها که دارای صدها هزار امتیاز کاربر هستند غالب شده است

در حالی که سایر برنامه ها برای عبور از آستانه 10،000 به سختی تلاش می کنند

ما می توانیم با حذف این برنامه های بسیار محبوب درهر ژانر ، تصویر بهتری داشته باشیم

و سپس میانگین ها را دوباره حساب کنیم ، اما این سطح از جزئیات را برای بعدا می گذاریم



است که غالب  امتیازات را از آن خود کرده است  Dictionary.com  و Bible   برنامه های مرجع به طور متوسط 74942 امتیاز کاربر دارند ، اما در واقع کتاب 


In [24]:
for app in ios_final:
    if app[-5] == 'Reference':
        print(app[1], ':', app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD â„¢ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokأ©mon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
Jishokun-Japanese English Dictionary & Translator : 0


کاری که می توانیم انجام دهیم این است که یک کتاب محبوب دیگر به برنامه تبدیل کنیم و علاوه بر نسخه خام کتاب ، ویژگی های مختلفی نیز به آن اضافه کنیم

می توان مواردی مانند نقل قول های روزانه از کتاب ، نسخه صوتی کتاب ، کویز های مربوط به کتاب و غیره را اضافه کرد

علاوه بر این ، می توانیم یک فرهنگ لغت را در برنامه تعبیه کنیم ، تا کاربران برای جستجوی لغات نیازی به خارج شدن از برنامه مان و جستجوی آن لغت در برنامه ای دیگر  نداشته باشند

<br>
<br>

به نظر می رسد این ایده با این واقعیت که اپ استور تحت سلطه برنامه های سرگرم کننده است ، کاملاً منطبق است و نشان می دهد 

که این بازار ممکن است اندکی با برنامه های سرگرم کننده اشباع شده باشد و 

بدان معناست که یک برنامه کاربردی ممکن است شانس بیشتری برای بولد شدن در میان تعداد زیادی از برنامه های اپ استور داشته باشد



<br>
<br>




ژانرهای دیگری که محبوب به نظر می رسند شامل آب و هوا ، کتاب ، غذا ، نوشیدنی و امور مالی هستند

به نظر می رسد ژانر کتاب با ایده برنامه ای که در بالا توضیح دادیم اندکی هم پوشانی دارد ، اما ژانرهای دیگر برای ما خیلی جالب به نظر نمی رسند


<br>
<br>



برنامه های هواشناسی - کاربران معمولاً زمان زیادی را در چنین برنامه هایی صرف نمی کنند و احتمال سودآوری از تبلیغات درون برنامه ای کم است

های غیر رایگان کند API  همچنین ، دریافت داده های لایو و  قابل اطمینان آب و هوا ممکن است ما را ملزم به اتصال برنامه خود به 


<br>
<br>


غذا و نوشیدنی - برنامه هایی مانند استارباکس ، دونکین دونات ، مک دونالد و غیره

بنابراین ساخت یک برنامه محبوب غذا و نوشیدنی نیاز به پخت و پز واقعی و خدمات دلیوری دارد ، که خارج از محدوده شرکت ما است.


<br>
<br>


برنامه های امور مالی - این برنامه ها شامل بانکداری ، پرداخت قبوض ، انتقال پول و غیره است 

ساختن یک برنامه مالی به دانش دامنه نیاز دارد و ما نمی خواهیم یک متخصص امور  مالی تنها برای ساخت یک برنامه استخدام کنیم.


حال بیایید بازار گوگل پلی را تحلیل کنیم.


## Most Popular Apps by Genre on Google Play

برای بازار گوگل پلی، داده های مربوط به  تعداد نصب ها را در واقع داریم ، بنابراین باید بتوانیم تصویر واضح تری در مورد محبوبیت ژانرها  بدست آوریم. 

با این حال ، تعداد نصب ها به اندازه کافی دقیق به نظر نمی رسند - می توانیم ببینیم که بیشتر مقادیر پایان باز دارند (+100 ، +1000 ، +5000 و غیره):

In [25]:
display_table(android_final, 5) # the Installs columns

1,000,000+ : 15.74200913242009
100,000+ : 11.518264840182649
10,000,000+ : 10.60502283105023
10,000+ : 10.205479452054794
1,000+ : 8.367579908675799
100+ : 6.952054794520548
5,000,000+ : 6.872146118721462
500,000+ : 5.5479452054794525
50,000+ : 4.7716894977168955
5,000+ : 4.486301369863014
10+ : 3.515981735159817
500+ : 3.2077625570776256
50,000,000+ : 2.28310502283105
100,000,000+ : 2.134703196347032
50+ : 1.9292237442922375
5+ : 0.7876712328767124
1+ : 0.5136986301369862
500,000,000+ : 0.273972602739726
1,000,000,000+ : 0.228310502283105
0+ : 0.045662100456621
0 : 0.01141552511415525


مشکلی که در این داده ها وجود دارد، دقیق نبودن آن است. به عنوان مثال ، ما نمی دانیم برنامه ای با +100000  نصب ، 100000 ، 200000 یا 350000 بار نصب شده است

با این حال ، ما برای اهداف خود به داده های خیلی دقیق نیاز نداریم - ما فقط می خواهیم بدانیم که چه  ژانرهایی بیشترین جذب کاربران را دارند و نیاز به دانستن تعداد دقیق کاربران نداریم

<br><br>

اعداد را همانطور که هست باقی خواهیم گذاشت، بدین معنا که برنامه ای با +100000 نصب 100000 بار و برنامه دیگری با +1000000 نصب 1000000 بار نصب شده است و غیره.

<br><br>


با این حال ، برای انجام محاسبات ، باید هر شماره نصب را به عدد اعشاری تبدیل کنیم - بدین معنا که باید کاما و علامت + را حذف کنیم 

در غیر این صورت تبدیل درست انجام نخواهد شد و با خطا مواجه می شویم. این کار را مستقیماً در حلقه زیر انجام خواهیم داد 

نیز محاسبه می کنیم (category) که میانگین تعداد نصب را برای هر ژانر

In [26]:
categories_android = freq_table(android_final, 1)

for category in categories_android:
    total = 0
    len_category = 0
    for app in android_final:
        category_app = app[1]
        if category_app == category:            
            n_installs = app[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total / len_category
    print(category, ':', avg_n_installs)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 654074.8271604938
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8329168.936170213
BUSINESS : 1712290.1474201474
COMICS : 859042.1568627451
COMMUNICATION : 38550548.03859649
DATING : 861409.5521472392
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11767380.952380951
EVENTS : 253542.22222222222
FINANCE : 1365500.4049079753
FOOD_AND_DRINK : 1951283.8055555555
HEALTH_AND_FITNESS : 4219697.055350553
HOUSE_AND_HOME : 1385541.463768116
LIBRARIES_AND_DEMO : 649314.0506329114
LIFESTYLE : 1447458.976676385
GAME : 15571586.690307328
FAMILY : 3716053.755274262
MEDICAL : 121161.87781350482
SOCIAL : 23628689.23275862
SHOPPING : 7103190.78680203
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3750580.6438356163
TRAVEL_AND_LOCAL : 14120454.07804878
TOOLS : 10902378.834454913
PERSONALIZATION : 5240358.986111111
PRODUCTIVITY : 16787331.344927534
PARENTING : 552875.1785714285
WEATHER : 5212877.101449275
VIDEO_PLAYERS : 24878048.860759493
NEWS_AND_MAGAZI

به طور میانگین، برنامه های ارتباطی بیشترین تعداد نصب را دارند: 38,456,119 

این تعداد توسط چند برنامه که بیش از یک میلیارد نصب دارند 

و چند برنامه دیگر با بیش از 100 و 500 میلیون نصب به شدت غالب شده است. (Hangouts و Messenger ، Skype ، Google Chrome ، Gmail ، WhatsApp ، Facebook ) 

In [27]:
for app in android_final:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Me

اگر همه برنامه های ارتباطی را که بیش از 100 میلیون نصب دارند حذف کنیم ، میانگین تقریباً ده برابر کاهش می یابد

In [28]:
under_100_m = []

for app in android_final:
    n_installs = app[5]
    n_installs = n_installs.replace(',', '')
    n_installs = n_installs.replace('+', '')
    if (app[1] == 'COMMUNICATION') and (float(n_installs) < 100000000):
        under_100_m.append(float(n_installs))
        
sum(under_100_m) / len(under_100_m)

3437620.895348837

می بینیم که با 24,727,872 نصب در رده دوم قرار دارد video players  همین الگو را برای دسته 

بازار را احاطه کرده اند MX Player و Youtube  ، Google Play Movies & TV  برنامه هایی مانند 

داریم (Google+ و Instagram ، Facebook ) این الگو برای برنامه های اجتماعی، جایی که غول هایی مانند 

یا برنامه های بهره وری ( و دیگر ویرایشگر های معروف عکس Google Photos   )برنامه های عکاسی 

تکرار می شود (Evernote و Google Calendar  ، Dropbox  ، Microsoft Word )

<br><br>


باز هم ، نگرانی اصلی این است که این ژانرها ممکن است محبوب تر از آنچه واقعاً هستند به نظر برسند. علاوه بر این ، به نظر می رسد که این دسته ها تحت سلطه چند غول هستند که رقابت در برابر آنها دشوار است
<br><br>


ژانر بازی بسیار محبوب به نظر می رسد ، اما قبلاً متوجه شده بودیم که این قسمت از بازار کمی اشباع به نظر می رسد ، بنابراین دوست داریم در صورت امکان یک ایده ی متفاوت ارائه دهیم.

<br><br>


به نظر می رسد كتاب ها و ژانر مرجع با تعداد میانگین  نصب 7,767,811 بسیار محبوب باشد. بررسی این موضوع به طور عمیق تر جالب به نظر می رسد

زیرا متوجه شدیم که این ژانر پتانسیل خوبی برای کار در اپ استور دارد و هدف ما پیشنهاد ژانری است که هم در اپ استور و هم در گوگل پلی پتانسیل سودآوری را داشته باشد.



بیایید نگاهی به برخی از برنامه های این ژانر و تعداد نصب آنها بیندازیم

In [29]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+

ژانر کتاب و مرجع شامل برنامه های متنوعی است: نرم افزاری برای پردازش و خواندن کتاب های الکترونیکی ، مجموعه های مختلف کتابخانه ها ، فرهنگ لغت ها ، آموزش های مربوط به برنامه نویسی یا زبان ها و غیره. به نظر می رسد هنوز تعداد اندکی از برنامه های محبوب میانگین را بالا کشیده اند.

In [30]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                            or app[5] == '500,000,000+'
                                            or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Audiobooks from Audible : 100,000,000+


با این حال ، به نظر می رسد که فقط چند برنامه بسیار محبوب وجود دارد ، بنابراین این بازار هنوز پتانسیل خود را دارد. بیایید سعی کنیم چند ایده برای برنامه مان براساس نوع برنامه هایی که از نظر محبوبیت در وسط قرار دارند (بین 1,000,000 تا 100,000,000 دانلود) بدست آوریم

In [31]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+

به نظر می رسد این دسته توسط نرم افزار های پردازش و خواندن کتاب های الکترونیکی و همچنین مجموعه های مختلف کتابخانه ها و دیکشنری ها احاطه شده است ، بنابراین ساخت یک برنامه مشابه از آنجا که رقابت قابل توجهی وجود دارد ، احتمالاً ایده خوبی نیست



همچنین متوجه می شویم که تعداد کمی برنامه حول کتاب قرآن ساخته شده است ، که نشان می دهد ساخت یک برنامه حول یک کتاب محبوب می تواند سودآور باشد. به نظر می رسد که گرفتن یک کتاب محبوب (یک کتاب جدید) و تبدیل آن به یک برنامه می تواند برای بازارهای گوگل پلی و اپ استور سودآور باشد.


## Conclusions

در این پروژه ، ما داده های مربوط به برنامه های تلفن همراه  اپ استور و گوگل پلی را با هدف یافتن ایده ای جهت ساخت یک برنامه که می تواند برای هر دو بازار سودآور باشد ، تجزیه و تحلیل کردیم

نتیجه گرفتیم که گرفتن یک کتاب مشهور (شاید یک کتاب جدید) و تبدیل آن به یک برنامه می تواند برای بازارهای گوگل پلی و اپ استور سودآور باشد. بازارها در حال حاضر پر از کتابخانه هستند ، بنابراین ما باید علاوه بر نسخه خام کتاب ، چندین ویژگی خاص اضافه کنیم. که ممکن است شامل نقل قول های روزانه از کتاب ، نسخه صوتی کتاب ، مسابقه های مربوط به کتاب ، تالار گفتگو که در آن افراد بتوانند درباره کتاب بحث کنند و غیره باشد.